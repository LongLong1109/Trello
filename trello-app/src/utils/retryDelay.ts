export const retryDelay = (attemptIndex: number) => Math.min(1000 * 2 ** attemptIndex, 30000)
