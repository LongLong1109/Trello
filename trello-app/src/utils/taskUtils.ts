//interface
import { ListTask, TaskListItem, Task } from '@/interfaces/Task'
import { TASK_PROPERTY_KEY, TaskPropertyValue } from '@/types/Task'

export const updateTaskProperty = (
  lists: ListTask[],
  listId: number,
  taskId: number,
  key: TASK_PROPERTY_KEY,
  value: TaskPropertyValue,
) => {
  return lists.map((list) => {
    const isTargetList = list.id === listId
    if (isTargetList) {
      return {
        ...list,
        tasks: list.tasks.map((task) => (task.id === taskId ? { ...task, [key]: value } : task)),
      }
    }
    return list
  })
}

export const removeTaskById = (lists: ListTask[], listId: number, taskId: number) => {
  return lists.map((list) => {
    const isTargetList = list.id === listId
    if (isTargetList) {
      return {
        ...list,
        tasks: list.tasks.filter((task) => task.id !== taskId),
      }
    }
    return list
  })
}

export const updateTaskTitle = (lists: ListTask[], selectedTask: TaskListItem, title: string) => {
  return lists.map((list) =>
    list.id === selectedTask.listId
      ? {
          ...list,
          tasks: list.tasks.map((task) =>
            task.id === selectedTask.id ? { ...task, title } : task,
          ),
        }
      : list,
  )
}

export const moveTask = (
  taskToMove: Task,
  updatedLists: ListTask[],
  targetListId: number,
  targetIndex: number,
) => {
  return updatedLists.map((list) => {
    if (list.id === targetListId) {
      const updatedTasks = [...list.tasks]
      updatedTasks.splice(targetIndex, 0, taskToMove)
      return { ...list, tasks: updatedTasks }
    }
    return list
  })
}
