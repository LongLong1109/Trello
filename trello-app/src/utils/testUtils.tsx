import { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { MantineProvider } from '@mantine/core'
const AllBrowserRouter = ({ children }: { children: React.ReactNode }) => {
  return <MantineProvider>{children}</MantineProvider>
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllBrowserRouter, ...options })

export * from '@testing-library/react'
export * from '@testing-library/user-event'
export { customRender as render }
