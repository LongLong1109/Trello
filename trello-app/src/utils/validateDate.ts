export const isDateOverdue = (date: Date) => {
  const currentDate = new Date()
  currentDate.setHours(0, 0, 0, 0)
  date.setHours(0, 0, 0, 0)

  const timeDifference = date.getTime() - currentDate.getTime()
  return timeDifference < 0
}

export const convertDateToString = (date: Date) => {
  const validDate = date ? new Date(date) : new Date()
  return validDate.toLocaleDateString()
}

export const formatDate = (date: Date): string => {
  if (!date || isNaN(date.getTime())) {
    return '---'
  }

  const options: Intl.DateTimeFormatOptions = { month: 'short', day: 'numeric' }
  return new Intl.DateTimeFormat('en-US', options).format(date)
}
