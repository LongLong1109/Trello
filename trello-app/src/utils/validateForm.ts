import { MESSAGES } from '@/constants/message'
import { REGEX_EMAIL, PASSWORD_MIN_LENGTH } from '@/constants/validateFields'

export const validateFirstName = (value: string) => {
  return value === '' ? MESSAGES.firstName : null
}

export const validateLastName = (value: string) => {
  return value === '' ? MESSAGES.lastName : null
}

export const validateEmail = (value: string) => {
  return REGEX_EMAIL.test(value) ? null : MESSAGES.email
}

export const validatePassword = (value: string) => {
  if (value === '') {
    return MESSAGES.passwordRequired
  }
  return value.length >= PASSWORD_MIN_LENGTH ? null : MESSAGES.password
}
