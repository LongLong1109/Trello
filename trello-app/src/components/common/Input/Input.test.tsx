import InputComponent from '.';

import { render } from '@/utils/testUtils';

describe('Input component', () => {
  const onChange = jest.fn();
  const renderInput = () => {
    return render(
      <InputComponent value='' placeholder='Email' pointer={true} onChange={onChange} />,
    );
  };

  it('should be matches snapshot', () => {
    const { asFragment } = renderInput();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly props', () => {
    const { getByPlaceholderText } = renderInput();
    expect(getByPlaceholderText('Email')).toBeInTheDocument();
  });
});
