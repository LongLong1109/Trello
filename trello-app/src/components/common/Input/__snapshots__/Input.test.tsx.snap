// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should be matches snapshot 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m_6c018570 mantine-Input-wrapper"
    data-pointer="true"
    data-variant="default"
    style="--input-left-section-pointer-events: none; --input-right-section-pointer-events: none;"
  >
    <input
      aria-invalid="false"
      class="m_8fb7ebe7 mantine-Input-input"
      data-variant="default"
      placeholder="Email"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`Input component should calls onChange when input value changes 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m_6c018570 mantine-Input-wrapper"
    data-pointer="true"
    data-variant="default"
    style="--input-left-section-pointer-events: none; --input-right-section-pointer-events: none;"
  >
    <input
      aria-invalid="false"
      class="m_8fb7ebe7 mantine-Input-input"
      data-testid="input-email"
      data-variant="default"
      placeholder="Email"
      value=""
    />
  </div>
</div>
`;
